pkg/model/volumes.go

```diff
@@ -0,0 +1,217 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/errors"
+)
+
+// HasLocalVolumes returns true if the manifest has local volumes
+func (dev *Dev) HasLocalVolumes() bool {
+	for _, v := range dev.Volumes {
+		if v.LocalPath != "" {
+			return true
+		}
+	}
+	return false
+}
+
+// HasRemoteVolumes returns true if the manifest has remote volumes
+func (dev *Dev) HasRemoteVolumes() bool {
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			return true
+		}
+	}
+	return false
+}
+
+// IsSyncFolder returns if path must be synched
+func (dev *Dev) IsSyncFolder(path string) (bool, error) {
+	found := false
+	for _, v := range dev.Volumes {
+		if v.LocalPath == "" {
+			continue
+		}
+		rel, err := filepath.Rel(v.LocalPath, path)
+		if err != nil {
+			return false, err
+		}
+		if strings.HasPrefix(rel, "..") {
+			continue
+		}
+		found = true
+		if rel != "." {
+			return false, nil
+		}
+	}
+	if found {
+		return true, nil
+	}
+	return false, errors.ErrNotFound
+}
+
+// PersistentVolumeEnabled returns true if persistent volumes are enabled for dev
+func (dev *Dev) PersistentVolumeEnabled() bool {
+	if dev.PersistentVolumeInfo == nil {
+		return true
+	}
+	return dev.PersistentVolumeInfo.Enabled
+}
+
+// PersistentVolumeSize returns the persistent volume size
+func (dev *Dev) PersistentVolumeSize() string {
+	if dev.PersistentVolumeInfo == nil {
+		return OktetoDefaultPVSize
+	}
+	if dev.PersistentVolumeInfo.Size == "" {
+		return OktetoDefaultPVSize
+	}
+	return dev.PersistentVolumeInfo.Size
+}
+
+// PersistentVolumeStorageClass returns the persistent volume storage class
+func (dev *Dev) PersistentVolumeStorageClass() string {
+	if dev.PersistentVolumeInfo == nil {
+		return ""
+	}
+	return dev.PersistentVolumeInfo.StorageClass
+}
+
+func (dev *Dev) validatePersistentVolume() error {
+	if dev.PersistentVolumeEnabled() {
+		return nil
+	}
+	if len(dev.Services) > 0 {
+		return fmt.Errorf("'persistentVolume.enabled' must be set to true to work with services")
+	}
+	if dev.HasRemoteVolumes() {
+		return fmt.Errorf("'persistentVolume.enabled' must be set to true to use remote volumes")
+	}
+	for _, v := range dev.Volumes {
+		result, err := dev.IsSyncFolder(v.LocalPath)
+		if err != nil {
+			return err
+		}
+		if !result {
+			return fmt.Errorf("'persistentVolume.enabled' must be set to true to use subfolders in the 'volumes' field")
```

Comment:
why do we need persistent volumes to do subfolders? now it's only needed if you want to share it with a service no?

Reply:
it is also required now, but we are not failing fast. subfolders use k8s volume mounts, and volume mounts require a volume to be defined